### ISSUE TITLE
[REVIEW]: PySPOD: A Python package for Spectral Proper Orthogonal Decomposition (SPOD)

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2862]
**Submitting author:** @mengaldo (<a href="http://orcid.org/0000-0002-0157-5477">Gianmarco Mengaldo</a>)
**Repository:** <a href="https://github.com/mengaldo/PySPOD" target ="_blank">https://github.com/mengaldo/PySPOD</a>
**Version:** v0.3
**Editor:** @eloisabentivegna
**Reviewer:** @albertonogueira, @joao-l-s-almeida, @jdmoorman
**Archive:** <a href="https://doi.org/10.5281/zenodo.4680581" target="_blank">10.5281/zenodo.4680581</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/5586d8b99091bc9ee97ec74070f52711/status.svg)](https://joss.theoj.org/papers/5586d8b99091bc9ee97ec74070f52711)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/5586d8b99091bc9ee97ec74070f52711"><img src="https://joss.theoj.org/papers/5586d8b99091bc9ee97ec74070f52711/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/5586d8b99091bc9ee97ec74070f52711/status.svg)](https://joss.theoj.org/papers/5586d8b99091bc9ee97ec74070f52711)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/mengaldo/PySPOD" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@albertonogueira & @joao-l-s-almeida & @jdmoorman, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @eloisabentivegna know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/mengaldo/PySPOD">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@mengaldo) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/mengaldo/PySPOD">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@mengaldo) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @albertonogueira, @joao-l-s-almeida, @jdmoorman it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2862 with the following error: 

 Can't find any papers to compile :-(

**Comment by eloisabentivegna** [comment]
@whedon generate pdf from branch paper/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch paper/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2862 with the following error: 

 Can't find any papers to compile :-(

**Comment by eloisabentivegna** [comment]
@whedon generate pdf from branch paper/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch paper/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2862 with the following error: 

 Can't find any papers to compile :-(

**Comment by eloisabentivegna** [comment]
@arfon, could you investigate why the paper is not found? I don't see any issues with the repository or the format. Thanks!

**Comment by mengaldo** [comment]
@eloisabentivegna I think that the branch is called papers/JOSS instead of paper/JOSS

**Comment by arfon** [comment]
@whedon generate pdf from branch papers/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch papers/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by eloisabentivegna** [comment]
> @eloisabentivegna I think that the branch is called papers/JOSS instead of paper/JOSS

🤦🏻‍♀️

Thanks!

**Comment by eloisabentivegna** [comment]
@whedon check references from branch papers/JOSS

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch papers/JOSS
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.1017/jfm.2018.283 may be a valid DOI for title: Spectral proper orthogonal decomposition and its relationship to dynamic mode decomposition and resolvent analysis
- 10.1175/mwr-d-18-0337.1 may be a valid DOI for title: Spectral empirical orthogonal function analysis of weather and climate data
- 10.1016/j.cpc.2018.11.009 may be a valid DOI for title: An efficient streaming algorithm for spectral proper orthogonal decomposition

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @albertonogueira, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @jdmoorman, please update us on how your review is going.

**Comment by jdmoorman** [comment]
I have started my review. 

Regarding scholarly effort, it is not clear whether the submission meets the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort). The majority of the content was committed all at once initially with about 3 additional weeks of development since the initial commit, judging from the [commit activity](https://github.com/mengaldo/PySPOD/graphs/commit-activity). As such, it is not clear whether the work represents at least three months of work for an individual. From a LOC perspective, the submission has around `3.2k` LOC total with `0.8k` in tests, `0.6k` in tutorials, and `1.6k` in the package. There is a large amount of code duplication between tests and tutorials, between the low RAM and low storage variants of the SPOD algorithm, and between the various post-processing/plotting functions. A refactor would cut the LOC by more than half.

**Comment by mengaldo** [comment]
Regarding the history of the package - please, note that this package comes from another (proprietary) package, and was deemed useful for some weather and climate applications. Therefore, I extracted the useful parts and pushed them into this repository making it available to the wider community. In regard to your comment about duplication, it is certainly true that some can be removed. I would be more than happy to hear your specific suggestions regarding a refactoring that could cut lines by more than half while keeping performance in place. Thank you for the feedback, very appreciated.

**Comment by jdmoorman** [comment]
Thanks for elaborating on the origins of the package! Can you confirm that no other authors/contributors to the original proprietary package should be included as authors for your submission?

**Comment by jdmoorman** [comment]
I have completed my review. 

The issues I created on the [target repository](https://github.com/mengaldo/PySPOD/issues) that are linked to this review issue are acceptance-blockers. The other issues I created are just suggestions to improve the project. Additionally,

- [ ] DOIs should be added to the references in the paper.

**Comment by Joao-L-S-Almeida** [comment]
@whedon start review

**Comment by whedon** [comment]
I'm sorry @Joao-L-S-Almeida, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by Joao-L-S-Almeida** [comment]
@whedon start review

**Comment by whedon** [comment]
I'm sorry @Joao-L-S-Almeida, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by Joao-L-S-Almeida** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by Joao-L-S-Almeida** [comment]
@whedon assign @Joao-L-S-Almeida as reviewer

**Comment by whedon** [comment]
I'm sorry @Joao-L-S-Almeida, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by danielskatz** [comment]
@Joao-L-S-Almeida - what are you trying to do?  Maybe I can help.

**Comment by danielskatz** [comment]
@whedon re-invite @Joao-L-S-Almeida as reviewer

(guessing maybe this is the issue)

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@joao-l-s-almeida please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by Joao-L-S-Almeida** [comment]
Thanks a lot @danielskatz, That was the issue.

**Comment by eloisabentivegna** [comment]
> @whedon re-invite @Joao-L-S-Almeida as reviewer
> 
> (guessing maybe this is the issue)

Thanks, @danielskatz!

**Comment by albertonogueira** [comment]
I am intrigued about how to compare traditional POD and SPOD. It should be explored in the corresponding paper of PySPOD.

**Comment by albertonogueira** [comment]
@albertonogueira and @Joao-L-S-Almeida completed their review. One issue was created on the target repository as an acceptance-blocker. The point concerning the comparison between POD and SPOD is a suggestion to improve the paper reading.

**Comment by eloisabentivegna** [comment]
> @albertonogueira and @Joao-L-S-Almeida completed their review. One issue was created on the target repository as an acceptance-blocker. The point concerning the comparison between POD and SPOD is a suggestion to improve the paper reading.

Thanks, @albertonogueira and @Joao-L-S-Almeida! Are the boxes you left unticked dependent on the author's responses to your suggestions?

Same question for @jdmoorman: are you waiting on the author's response before you tick off the remaining items on your list?

**Comment by albertonogueira** [comment]
Yes, @eloisabentivegna. The unticked boxes correspond to issues created on the target repository. Once the issues are addressed, the boxes will be ticked.

**Comment by jdmoorman** [comment]
> Yes, @eloisabentivegna. The unticked boxes correspond to issues created on the target repository. Once the issues are addressed, the boxes will be ticked.

Same.

**Comment by mengaldo** [comment]
> Thanks for elaborating on the origins of the package! Can you confirm that no other authors/contributors to the original proprietary package should be included as authors for your submission?

@jdmoorman Yes, I confirm that! 
I would want to add Romit Maulik as an author instead of a contributor. I will update the author list accordingly.

**Comment by mengaldo** [comment]
@eloisabentivegna I will respond to the various issues raised over the next few weeks, as it is a quite busy period. Hope this is OK.

**Comment by eloisabentivegna** [comment]
> @eloisabentivegna I will respond to the various issues raised over the next few weeks, as it is a quite busy period. Hope this is OK.

Certainly!

**Comment by eloisabentivegna** [comment]
@mengaldo, do you have an update regarding the reviewers requests, and especially when you expect to be able to respond?

**Comment by mengaldo** [comment]
@eloisabentivegna I am planning to work on it second half of February.

**Comment by eloisabentivegna** [comment]
@mengaldo, that's OK. Please keep us posted if you encounter any issues with the requests.

**Comment by eloisabentivegna** [comment]
@mengaldo, have you had a chance to work on the reviews yet? Could you post an update?

**Comment by mengaldo** [comment]
@eloisabentivegna yes, I have almost completed all the points raised. I am now looking at the few remaining ones. I believe that by the end of March the process will be complete.

**Comment by eloisabentivegna** [comment]
@mengaldo, that's OK, but let's take the end of March as an official hard deadline as this task has been open for over two months. If you think you might not be unable to complete the process by the end of the month, please let us know promptly.

**Comment by mengaldo** [comment]
@eloisabentivegna Thank you for your patience. I have now completed the review.

**Comment by eloisabentivegna** [comment]
@albertonogueira and @Joao-L-S-Almeida, could you check whether you're happy with the results?

**Comment by albertonogueira** [comment]
Yes, I am Ok with all the reviews.

**Comment by albertonogueira** [comment]
I recommend this submission for publication.

**Comment by Joao-L-S-Almeida** [comment]
I am in agreement with the reviews and I recommend this submission. Thank you for the efforts.

**Comment by eloisabentivegna** [comment]
Thanks, @albertonogueira and @Joao-L-S-Almeida! Your insight and suggestions are very appreciated.

@jdoorman, from your checklist it would seem you are done with the review? Could you confirm and issue your recommendation?

**Comment by jdmoorman** [comment]
I am finished with my review, I recommend this submission for publication.

**Comment by eloisabentivegna** [comment]
Thanks, @jdmoorman!

I am happy to start the pre-publication phase for this submission. Congratulations, @mengaldo!

**Comment by eloisabentivegna** [comment]
@whedon generate pdf from branch paper/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch paper/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch papers/JOSS
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2862 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by eloisabentivegna** [comment]
@whedon generate pdf from branch papers/JOSS

**Comment by eloisabentivegna** [comment]
@mengaldo, are the DOI suggestions above valid? If so, could you add them to your `paper.bib`?

A few other style suggestions:
1) Line 12: "otherwise be not predictable" -> "not otherwise be predictable"
2) Line 13: "dynamical system theory" -> "dynamical systems theory"
3) Line 24: introduce the acronym SVD before using it (perhaps on line 15?)
4) Line 51: "their needs" -> "their own needs"
5) Line 53: "fluidmechanics" -> "fluid mechanics"
6) Lines 57, 61: `autoref` should not be preceded by "figure".
7) Line 67: please remove the "Acknowledgements" header if this section is empty

**Comment by mengaldo** [comment]
@eloisabentivegna Thank you! Note that as I pointed out in a comment earlier, I also would like to add Romit Maulik as an author. I am going to make the changes above and come back to you shortly.

**Comment by mengaldo** [comment]
@whedon generate pdf from branch papers/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch papers/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by mengaldo** [comment]
@whedon generate pdf from branch papers/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch papers/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by mengaldo** [comment]
@whedon generate pdf from branch papers/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch papers/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by mengaldo** [comment]
@whedon generate pdf from branch papers/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch papers/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by mengaldo** [comment]
@eloisabentivegna All corrections you suggested done + DOI added. I think that it is good to go on my end.

**Comment by eloisabentivegna** [comment]
@whedon check references from branch papers/JOSS

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch papers/JOSS
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1017/jfm.2018.283 is OK
- 10.1175/mwr-d-18-0337.1 is OK
- 10.1016/j.cpc.2018.11.009 is OK
- 10.1038/s41561-017-0033-0 is OK

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.2514/1.J058809 is INVALID because of 'https://doi.org/' prefix
```

**Comment by eloisabentivegna** [comment]
@mengaldo, could you fix the last DOI, and once you're done, generate a tagged release and code archive as per https://joss.readthedocs.io/en/latest/submitting.html#the-review-process?

Thanks!

**Comment by mengaldo** [comment]
@eloisabentivegna Should I create the the tagged release for the main branch and not for the JOSS branch, I suppose. Correct?

Thank you!

**Comment by mengaldo** [comment]
@eloisabentivegna Just completed the above tasks. Here it is: https://doi.org/10.5281/zenodo.4680581

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by kyleniemeyer** [comment]
@whedon generate pdf from branch papers/JOSS

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch papers/JOSS. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by kyleniemeyer** [comment]
Hi @mengaldo, I am the AEIC on duty this week looking to help get this submission past the finish line.

There are a few remaining issues with your article that need addressing:
- [ ] In the author name list, the formatting we use is "Given Middle-initial Surname"; it looks like yours are reversed. Could you review and correct this? Normally we also use written-out names, with only an initial for middle name (if you have one).
- [ ] There are a handful of citations to software that do not use the appropriate bibtex fields; could you move the author names to the `author` field for the various SPOD code citations?
- [ ] It also looks like some citations throughout the paper do not use the appropriate syntax, in the third and fourth paragraphs. The options include:
  - `@author:2001`  ->  "Author et al. (2001)"
  - `[@author:2001]` -> "(Author et al., 2001)"
  - `[@author1:2001; @author2:2001]` -> "(Author1 et al., 2001; Author2 et al., 2002)"

For example, the "(Lumley, 2007), (Towne et al. 2018)" should instead appear as "(Lumley, 2007; Towne et al., 2018)" with the appropriate command of `[@lumley1970;@towne2017]`. There are a few other cases to fix as well.

Could you make these edits, and let us know when you have?

**Comment by kyleniemeyer** [comment]
@whedon set 10.5281/zenodo.4680581 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4680581" target="_blank">10.5281/zenodo.4680581</a> is the archive.

**Comment by mengaldo** [comment]
@whedon generate pdf from branch main

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch main. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by mengaldo** [comment]
@kyleniemeyer Thank you for the help! These should be all fixed. The only issue seems to be related to the url of one of the references. Also, please use the main branch to compile the paper now, as I moved everything there, and papers/JOSS will be removed. Thank you for handling the publication process!

**Comment by eloisabentivegna** [comment]
> Hi @mengaldo, I am the AEIC on duty this week looking to help get this submission past the finish line.

Thanks, @kyleniemeyer!

**Comment by kyleniemeyer** [comment]
Hi @mengaldo, I still saw some minor issues (or maybe I'm just very particular), which I fixed in https://github.com/mengaldo/PySPOD/pull/14 - can you merge this?

**Comment by mengaldo** [comment]
@kyleniemeyer Thank you, just did.

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02862/joss.02862/10.21105.joss.02862.pdf) :page_facing_up: :point_left:

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1017/jfm.2018.283 is OK
- 10.1175/mwr-d-18-0337.1 is OK
- 10.1016/j.cpc.2018.11.009 is OK
- 10.2514/1.J058809 is OK
- 10.1038/s41561-017-0033-0 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2240

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2240, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1382906484930052096) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2241
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02862](https://doi.org/10.21105/joss.02862)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats @mengaldo on your article's publication in JOSS!

Many thanks to  @albertonogueira, @Joao-L-S-Almeida, and @jdmoorman for reviewing this, and @eloisabentivegna for editing.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02862/status.svg)](https://doi.org/10.21105/joss.02862)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02862">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02862/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02862/status.svg
   :target: https://doi.org/10.21105/joss.02862
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02862/status.svg)](https://doi.org/10.21105/joss.02862)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
